File: HitsterApp/Services/PlayerService.js
import { collection, addDoc } from "firebase/firestore";
import { db } from "../firebase";

export async function addPlayerToGame(gameId, username) {
  const player = {
    username: username,
    score: 0,
    tokens: 0,
    timeline: [],
  };

  const docRef = await addDoc(
    collection(db, "games", gameId, "players"),
    player,
  );

  return docRef.id;
}

File: HitsterApp/Services/PlayerService.cs

using Google.Cloud.Firestore;

namespace HitsterApp.Services;
public class PlayerService
{
    private readonly FirestoreDb _db;

    public PlayerService(FirestoreService firestoreService)
    {
        _db = firestoreService.Db;
    }

    public async Task AddPlayer(string gameId, string username)
    {
        CollectionReference players =
            _db.Collection("games")
                .Document(gameId)
                .Collection("players");

        var player = new
        {
            Username = username,
            Score = 0
        };

        await players.AddAsync(player);
    }
}